stdlib/types/paths.nix

```diff
@@ -0,0 +1,79 @@
+{ lambdas, strings, ... }:
+{
+  isType = builtins.isPath;
+
+  # FIXME: is this correct?
+  toType = str: /. + toString str;
```

Comment:
This is correct when `str` is an absolute pathname.

For relative pathnames the result might not be intuitive, as paths will always be interpreted relative to `/`. 
E.g. let's imagine the simple case where there is a file `/tmp/test.nix` that calls `stdlib.types.paths.toType "./derp"`, then the user might reasonably assume the result to be `/tmp/derp` as that would correspond with way Nix treats paths. As there doesn't seem to be a way to capture the source file where the input string originates, it's probably better to add an assertion and only admit absolute pathnames here.

Reply:
Good call. Do you mind creating an issue on https://github.com/numtide/nix-stdlib?

Reply:
https://github.com/numtide/nix-stdlib/issues/2